app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, number: int | float) -> Distance:
+        if isinstance(number, Distance):
+            return Distance(self.km + number.km)
+        return Distance(self.km + number)
+
+    def __iadd__(self, number: int | float) -> Distance:
+        if isinstance(number, Distance):
+            self.km += number.km
+        else:
+            self.km += number
+        return self
+
+    def __mul__(self, number: int | float) -> Distance:
+        return Distance(self.km * number)
+
+    def __truediv__(self, number: int | float) -> Distance:
+        return Distance(round((self.km / number), 2))
+
+    def __lt__(self, number: int | float) -> bool:
+        if isinstance(number, Distance):
+            return self.km < number.km
+        return self.km < number
+
+    def __gt__(self, number: int | float) -> bool:
+        if isinstance(number, Distance):
+            return self.km > number.km
+        return self.km > number
+
+    def __eq__(self, number: int | float) -> bool:
+        if isinstance(number, Distance):
+            return self.km == number.km
+        return self.km == number
+
+    def __le__(self, number: int | float) -> bool:
+        if isinstance(number, Distance):
+            return self.km <= number.km
+        return self.km <= number
+
+    def __ge__(self, number: int | float) -> bool:
+        if isinstance(number, Distance):
+            return self.km >= number.km
+        return self.km >= number
```

Comment:
you can do this in one line using previously created magic methods. This will also allow you to get rid of those type checks